### PR TITLE
Prepare Model\Notify for include/enotify.php (Part 1)

### PR DESCRIPTION
**Pull request by nupplaphil**
Just small enhancements - preparation for more :) :
-  Move "Notify::getTab()" to notification module since it's presentation layer logic
- <strike>Introduce "order" and "limit" argument instead of "param" array for BaseRepository (more explicit, less implicit :-) )</strike>
- Make Repositories more Dependency Injectable (avoid using `DI::` inside already dynamic classes)

### CODE REVIEW

**Comment by annando** [comment]
Using `$param` does make more sense, since the query parameters can contain more than order or select range. Also "group_by" is supported - and possibly some stuff more in the future.

**Comment by nupplaphil** [comment]
Hmkay.. But I think that a repo is more business context orientet... So I would see `Repo::select()` as generic, last possibility if no other context orientet methods (getById, getForBusinessContext, ..) work.

Otherwise we could have sticked to the DB calls as well without a benefit introducing repos :)

**Comment by annando** [comment]
Well, for me, the `select` function is always the regular database call, possibly only enhanced with some bells and whistles, just like with the item class. I don't see any benefit in restricting yourself.

We could also always add some more functionality to the underlying database functions that wouldn't then be available.

But, well, I'm just the database guy not understanding the benefit of some abstraction layer that prevents me from calling the database directly.

**Comment by nupplaphil** [comment]
So the method name is really misleading.. At least I think that's maybe a reason for our discussion.. But I'll revert it, I understand that `select()` will always be in context of a DB select..

The main advantage of repositories is that they are a trustable, "bulletproof" source for typesafe items of the DB. So passing a `group_by` as argument f.e. would brick this concept because it will possibly group more items of the DB into one result, which makes it hard to determine if the result of the repo call is one of more results per model result.

 I'm really sorry that it seems like I offended you with my explanation about the design discussions between my companies DB-experts and JAVA-devs in one of our discussions before :( .. You're **not** just a DB guy :( .. And I appreciate our discussions here so I have to rethink my decisions and alter it in case I made it too complicated..

**Comment by annando** [comment]
Well, in fact I'm a database centric guy. I creating applications around the the database and the tables. The code is (in my eyes) "just" some helper stuff for the database. I do understand the background of this "repository" thing, but I have the feeling as if it is solving problems that we don't have.

Concerning the method name: Yes, I guess that another name would be a good idea.

**Comment by nupplaphil** [comment]
:D ... And I think in Business Cases / Business Processes and how we can therefore interact properly with the DB ... Similar result with a different approach ;-)

File: src/Repository/Introduction.php [hunk 1]
```diff
@@ -35,26 +35,28 @@ public function selectFirst(array $condition)
...
+	 * @param array $order
 	 * @param int|null $max_id
 	 * @param int|null $since_id
-	 * @param int $limit
+	 * @param int|array $limit
 	 * @return Collection\Introductions
 	 * @throws \Exception
 	 */
-	public function selectByBoundaries(array $condition = [], array $params = [], int $max_id = null, int $since_id = null, int $limit = self::LIMIT)
+	public function selectByBoundaries(array $condition = [], array $order = [], int $max_id = null, int $since_id = null, int $limit = self::LIMIT)
```
Comment (MrPetovan) [inline src/Repository/Introduction.php]: Did you update all calls to `selectByBoundaries`?

**Comment by nupplaphil** [comment]
I reverted the whole "select()/selectFirst()" changes, so it's even smaller now :)

**Comment by nupplaphil** [comment]
These constants aren't part of the model, they are part of the presentation layer like the only situation where we use it `getTabs()` => module
The general constant which kind of notifies exist is inside the model class.

**Comment by nupplaphil** [comment]
It's one part how I like to move the `include/enotify.php` inside `src/`, where `Notify` currently exists .. It seems like the `Model\Notify` class isn't **one** model .. it depends on what you do with it (`setSeen()` vs. `getIntroList()`)... The first is a classic model call, the second is a repository call of `introduction`, but extended with contact infos ...

**Pull request merged**